frontend/app/src/components/display/DateDisplay.vue

```diff
@@ -57,35 +55,21 @@ const showTooltip = computed(() => {
 const splittedByMillisecondsPart = computed(() =>
   get(formattedDate).split('.')
 );
+
+const { copy, copied } = useCopy(formattedDate);
 </script>
 
 <template>
-  <span>
-    <RuiTooltip
-      :popper="{ placement: 'top' }"
-      open-delay="400"
-      :disabled="!showTooltip"
+  <span :class="{ blur: !shouldShowAmount }" @click="copy()">
+    <CopyTooltip
+      class="items-baseline"
+      :copied="copied"
+      :tooltip="showTooltip ? formattedDateWithTimezone : null"
     >
-      <template #activator="{ on, attrs }">
-        <span
-          class="date-display whitespace-none"
-          :class="{ [css.blur]: !shouldShowAmount }"
-          v-bind="attrs"
-          v-on="on"
-        >
-          <span>{{ splittedByMillisecondsPart[0] }}</span>
-          <span v-if="splittedByMillisecondsPart[1]" class="text-[0.625rem]">
-            .{{ splittedByMillisecondsPart[1] }}
-          </span>
-        </span>
-      </template>
-      <span> {{ formattedDateWithTimezone }} </span>
-    </RuiTooltip>
+      <span>{{ splittedByMillisecondsPart[0] }}</span>
```

Comment:
I guess we can probably remove this wrapping span right?
Do you think it would be possible to remove the wrapping span too? would it work if we move the blur and copy to the `CopyTooltip` itself?

Reply:
Since this will affect history let's make sure that we use the minimum amount of DOM elements required.

Reply:
updated